pkg/controllers/gameservers/controller.go

```diff
@@ -369,13 +393,69 @@ func (c *Controller) syncGameServerDeletionTimestamp(gs *carrierv1alpha1.GameSer
 	return gs, errors.Wrap(err, "error removing finalizer for GameServer")
 }
 
+// tryAllocatePorts try to allocate port for GameServer, whose port policy is Dynamic.
+func (c *Controller) tryAllocatePorts(gs *carrierv1alpha1.GameServer) (*carrierv1alpha1.GameServer, error) {
+	if len(gs.Spec.Ports) == 0 {
+		return gs, nil
+	}
+	if IsDynamicPortAllocated(gs) {
+		return gs, nil
+	}
+	allocateType := getAllocateType(gs)
+	// only allow use multi ports or directly use port range.
+	gsCopy := gs.DeepCopy()
+	var (
+		ports []int
+		err   error
+	)
+	switch allocateType {
+	case "":
```

Comment:
For `LoadBalaner` policy, or `Spec.Ports == nil `, we can skip this.

Reply:
Done